CMakeLists.txt

```
@@ -156,6 +156,27 @@ ELSE ()
 ENDIF ()
 OPTION(WITH_KTLS "use Kernel TLS" ${WITH_KTLS_DEFAULT})
 
+# Find out if system provides libynl
+CMAKE_PUSH_CHECK_STATE()
+FIND_LIBRARY(SYSTEM_YNL_LIB "ynl")
+IF (SYSTEM_YNL_LIB)
+    SET(CMAKE_REQUIRED_LIBRARIES ${SYSTEM_YNL_LIB})
+ENDIF()
+CHECK_C_SOURCE_COMPILES("
+#include <ynl/ynl.h>
+int main(void) {
+return YNL_ERROR_NONE;
```

```suggestion
    ynl_ntf_free(NULL);
    return 0;
```
We want the linker to fail when libynl is not available. To achieve that, we have to use to a symbol defined in libynl.a rather than using a macro that is resolved by looking solely at the header file.